### ISSUE TITLE
Training time

### ISSUE THREAD

**Issue by jizongFox** [HobbitLong/SupContrast#15]
Hi @HobbitLong 
Thanks for the wonderful repo. 

I am currently using a machine with 2 gpus (1080) to reproduces the SimCLR and SupConstrast on cifar10. 
Could you please tell me how much time it would take for both methods until their convergence? I had some terrible experience of waiting for weeks for some parallel approaches. 

Thanks for the information. 
Jizong

**Comment by HobbitLong** [comment]
Hi, @jizongFox,

IIRC, it requires ~5 hours with 8 V100 GPUs to train 1000 epochs. You could also see the print out from each epoch to compute the estimated time of arrival.

**Comment by jizongFox** [comment]
It took me 2 days and a half. To give everyone a reference.

**Comment by lclkent** [comment]
> It took me 2 days and a half. To give everyone a reference.

Hi, thanks for sharing. How many epoch does it take to converge? And what was the loss value when it converges?

**Comment by ivanpanshin** [comment]
It took me about 20 hours with 1x2080TI with FP16 to train SupCon. My final scores were: 94.08 for CE and 95.47 for SupCon. Since I have 1 card instead of 8, I just lowered BS and LR by 8 times.

**Comment by wjun0830** [comment]
Is there any way of making loss function to use multiple GPU devices?
There is a serious bottleneck in calculating the SupConLoss.

**Comment by dvdsosa** [comment]
> It took me about 20 hours with 1x2080TI with FP16 to train SupCon. My final scores were: 94.08 for CE and 95.47 for SupCon. Since I have 1 card instead of 8, I just lowered BS and LR by 8 times.

Hi! Apart from decreasing BS and LR by 8 times, did you need to increase the epochs about 8 times to get those results? Thanks.

**Comment by HobbitLong** [comment]
> Is there any way of making loss function to use multiple GPU devices? There is a serious bottleneck in calculating the SupConLoss.

I got a new implementation [here](https://github.com/google-research/syn-rep-learn/blob/main/StableRep/models/losses.py#L49) that can be easily repurposed as DDP SupCon loss that can be used with multi-node & multi-gpu cases.
